docs/content/docs/ops/state/unaligned_checkpoints.md

```diff
@@ -0,0 +1,118 @@
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# Unaligned checkpoints
+
+Starting with Flink 1.11, checkpoints can be unaligned.
+[Unaligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#unaligned-checkpointing) 
+contain in-flight data (i.e., data stored in buffers) as part of the checkpoint state, allowing
+checkpoint barriers to overtake these buffers. Thus, the checkpoint duration becomes independent of
+the current throughput as checkpoint barriers are effectively not embedded into the stream of data
+anymore.
+
+You should use unaligned checkpoints if your checkpointing durations are very high due to
+backpressure. Then, checkpointing time becomes mostly independent of the end-to-end latency. Be
+aware unaligned checkpointing adds to I/O to the state backends, so you shouldn't use it when the
+I/O to the state backend is actually the bottleneck during checkpointing.
+
+### Alignment timeout
+
+After enabling unaligned checkpoints, you can also specify the alignment timeout programmatically:
+
+```java
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+env.getCheckpointConfig().setAlignmentTimeout(Duration.ofSeconds(30));
+```
+
+or in the `flink-conf.yml` configuration file:
+
+```
+execution.checkpointing.alignment-timeout: 30 s
+```
+
+When activated, each checkpoint will still begin as an aligned checkpoint, but if the alignment time
+for some subtask exceeds this timeout, then the checkpoint will proceed as an unaligned checkpoint.
+
+## Limitations
+
+### Concurrent checkpoints
+
+Flink currently does not support concurrent unaligned checkpoints. However, due to the more
+predictable and shorter checkpointing times, concurrent checkpoints might not be needed at all.
+However, savepoints can also not happen concurrently to unaligned checkpoints, so they will take
+slightly longer.
+
+### Interplay with watermarks
+
+Unaligned checkpoints break with an implicit guarantee in respect to watermarks during recovery.
+Currently, Flink generates the watermark as the first step of recovery instead of storing the latest
+watermark in the operators to ease rescaling. In unaligned checkpoints, that means on recovery,
+**Flink generates watermarks after it restores in-flight data**. If your pipeline uses an
+**operator that applies the latest watermark on each record** will produce **different results** than for
+aligned checkpoints. If your operator depends on the latest watermark being always available, the
+workaround is to store the watermark in the operator state. In that case, watermarks should be
+stored per key group in a union state to support rescaling.
+
+### Certain connections are not checkpointed
```

Comment:
`connection` -> `data distribution pattern`? I think connection is very technical and not easy to see for users.

Reply:
Hmm... personally I feel otherwise ;) `data distribution pattern` sounds more technical than `connection` to me. I might be wrong though. Would you replace connection/connections everywhere? Would e.g. change `Broadcast connection` to `Broadcast distribution pattern`?

Reply:
For me connection = channels, but I see that it may be different for different readers. Let's hear from @pnowojski about his view. Might be fine to leave as is. Alternative: connection type?